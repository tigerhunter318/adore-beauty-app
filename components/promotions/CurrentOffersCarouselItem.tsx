import React from 'react'
import { StyleSheet } from 'react-native'
import { formatPromotionData } from './utils/helpers'
import Container from '../ui/Container'
import Type from '../ui/Type'
import ResponsiveImage from '../ui/ResponsiveImage'
import theme from '../../constants/theme'
import ImageSize from '../../constants/ImageSize'

const styles = StyleSheet.create({
  item: {
    borderRadius: 27,
    alignItems: 'center',
    justifyContent: 'center'
  },
  imageContainer: {
    width: 160,
    height: 160,
    alignItems: 'center',
    justifyContent: 'center'
  }
})

type CurrentOffersCarouselItemProps = {
  item: {}
  index: number
  onPress: ({ item }: { item: object }) => void
  height?: number
  imageSize?: number
}

const CurrentOffersCarouselItem = ({
  item,
  index,
  onPress,
  height = 380,
  imageSize = 160
}: CurrentOffersCarouselItemProps) => {
  const { brandLogo, image, title } = formatPromotionData(item)
  const imageUrl = image || brandLogo
  const { width: imageWidth, height: imageHeight } = ImageSize.promotion.thumbnail

  return (
    <Container ph={1} style={[styles.item, { height }]} key={index} onPress={() => onPress({ item })}>
      {imageUrl ? (
        <Container style={styles.imageContainer}>
          <ResponsiveImage
            src={imageUrl}
            width={imageWidth}
            height={imageHeight}
            displayWidth={imageSize}
            useAspectRatio
          />
        </Container>
      ) : (
        <Container
          style={{ width: imageSize, height: imageSize }} // image placeholder
        />
      )}
      <Type semiBold letterSpacing={1.5} center mt={2} style={{ height: 70 }} numberOfLines={4}>
        {title}
      </Type>
      <Container>
        <Type heading size={12} pt={2} color={theme.lighterBlack}>
          View Offer
        </Type>
        <Container style={{ borderBottomWidth: 1, borderBottomColor: theme.darkGray, paddingTop: 5 }} />
      </Container>
    </Container>
  )
}

export default CurrentOffersCarouselItem
