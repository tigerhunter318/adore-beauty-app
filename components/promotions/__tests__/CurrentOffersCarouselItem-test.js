import React from 'react'
import renderer from 'react-test-renderer'
import CurrentOffersCarouselItem from '../CurrentOffersCarouselItem'

const mockData = require('./__mocks__/CurrentOffersCarouselItem.json')

it(`CurrentOffersCarouselItem renders content`, () => {
  const tree = renderer.create(<CurrentOffersCarouselItem item={mockData} />).toJSON()

  expect(tree).toMatchSnapshot()
})
