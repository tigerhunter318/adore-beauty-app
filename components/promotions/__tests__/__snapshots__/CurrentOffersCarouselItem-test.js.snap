// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentOffersCarouselItem renders content 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 27,
      "height": 380,
      "justifyContent": "center",
      "opacity": 1,
      "paddingLeft": 10,
      "paddingRight": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignItems": "center",
          "height": 160,
          "justifyContent": "center",
          "width": 160,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "height": "auto",
                "width": "100%",
              },
              Object {
                "aspectRatio": 1,
                "width": 160,
              },
            ],
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          imageProps={Object {}}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://www.adorebeauty.com.au/pim_brand_logo/000/001/364/Boost_Lab_Logo_Stacked_Black.png?w=210&h=210&fmt=webp",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <Text
    allowFontScaling={false}
    mt={2}
    numberOfLines={4}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "letterSpacing": 1.5,
          "marginTop": 20,
          "textAlign": "center",
        },
        Object {
          "height": 70,
        },
      ]
    }
  >
    Boost Lab Hydro Boost Serum 10ml*
  </Text>
  <View
    style={
      Array [
        Object {},
        Object {},
      ]
    }
  >
    <Text
      allowFontScaling={false}
      pt={2}
      style={
        Array [
          Object {
            "color": "#4A4A4A",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 12,
            "paddingTop": 20,
            "textTransform": "uppercase",
          },
          undefined,
        ]
      }
    >
      View Offer
    </Text>
    <View
      style={
        Array [
          Object {},
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "paddingTop": 5,
          },
        ]
      }
    />
  </View>
</View>
`;
